File: lab 1 scala/src/zadanie3.scala


class zadanie3 {
  
  
  def dwaRowne[A](xss: List[A]):Boolean=
   if (xss==Nil) false
	 else if (xss.tail == Nil) false
	 else if (xss.head == xss.tail.head ) true 
	 else dwaRowne(xss.tail)

val x1 = 1::2::3::4::Nil
dwaRowne(x1)
  
  
  
  
  
  
  
  
  
  
}

File: lab 1 ocaml/zadanie3.ml
(*zadanie 3*)
let rec dwaRowne xss=
	if xss=[] then false
	else if List.tl xss = [] then false
	else if List.hd xss = List.hd(List.tl xss) then true 
	else dwaRowne (List.tl xss);;


dwaRowne[1;2;3;4];;
	
	